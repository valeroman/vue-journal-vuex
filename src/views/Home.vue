<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png">

    <h1>Hola Mundo</h1>
    <button @click="goToDaybook" class="btn btn-primary">primary</button>
    <button @click="goToDaybook" class="btn btn-secondary">secondary</button>
    <button @click="goToDaybook" class="btn btn-success">Success</button>
  </div>
</template>

<script>
export default {
  methods: {
    goToDaybook() {
      this.$router.push({ name: 'no-entry' })
    }
  }
}
</script>

