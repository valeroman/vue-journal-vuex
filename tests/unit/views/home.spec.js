import Home from '@/views/Home'
import { shallowMount } from '@vue/test-utils'
describe('Pruebas en el Home View', () => {

    test('debe de renderizar el componente correctamente ', () => {

        const wrapper = shallowMount(Home)
        expect(wrapper.html()).toMatchSnapshot()

    })

    test('hacer click en un boton debe de redireccionar a no-entry ', () => {

        const mockRouter = {
            push: jest.fn()
        }

        const wrapper = shallowMount(Home, {
            global: {
                mocks: {
                    $router: mockRouter
                }
            }
        })

        wrapper.find('button').trigger('click')

        expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' })
    })

})