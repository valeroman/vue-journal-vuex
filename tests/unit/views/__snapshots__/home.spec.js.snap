// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Home View debe de renderizar el componente correctamente  1`] = `
<div class="home"><img alt="Vue logo" src="">
  <h1>Hola Mundo</h1><button class="btn btn-primary">primary</button><button class="btn btn-secondary">secondary</button><button class="btn btn-success">Success</button>
</div>
`;
